common/src/float_ops.rs

```diff
@@ -83,6 +83,23 @@ pub fn is_integer(v: f64) -> bool {
     (v - v.round()).abs() < std::f64::EPSILON
 }
 
+// Formats floats into Python style exponent notation, by first formatting in Rust style
+// exponent notation (`1.0000e0`), then convert to Python style (`1.0000e+00`).
+pub fn format_float_as_exponent(precision: usize, magnitude: f64) -> String {
+    if magnitude.is_nan() {
```

Comment:
I suggest to check `magnitude.is_finite()` first. Which is more common case but for now experiencing 2 branches for every formatting.